extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveCdiProcessor.java

```diff
@@ -0,0 +1,88 @@
+package io.quarkus.hibernate.reactive.deployment;
+
+import static io.quarkus.hibernate.orm.deployment.ClassNames.IMPLEMENTOR;
+import static io.quarkus.hibernate.orm.deployment.ClassNames.MUTINY_SESSION_FACTORY;
+
+import java.util.Arrays;
+import java.util.List;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Default;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.reactive.mutiny.Mutiny;
+import org.jboss.jandex.ClassType;
+import org.jboss.jandex.DotName;
+
+import io.quarkus.agroal.spi.JdbcDataSourceBuildItem;
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.SyntheticBeanBuildItem;
+import io.quarkus.deployment.Capabilities;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.hibernate.orm.deployment.ImpliedBlockingPersistenceUnitTypeBuildItem;
+import io.quarkus.hibernate.orm.deployment.PersistenceUnitDescriptorBuildItem;
+import io.quarkus.hibernate.orm.runtime.JPAConfig;
+import io.quarkus.hibernate.orm.runtime.PersistenceUnitUtil;
+import io.quarkus.hibernate.reactive.runtime.HibernateReactiveRecorder;
+
+public class HibernateReactiveCdiProcessor {
+
+    private static final List<DotName> MUTINY_SESSION_FACTORY_EXPOSED_TYPES = Arrays.asList(
+            MUTINY_SESSION_FACTORY, IMPLEMENTOR);
+
+    // These beans must be initialized at runtime because their initialization
+    // depends on runtime configuration (to activate/deactivate a persistence unit)
+    @Record(ExecutionTime.RUNTIME_INIT)
+    @BuildStep
+    void generateDataSourceBeans(
+            HibernateReactiveRecorder recorder,
+            List<PersistenceUnitDescriptorBuildItem> persistenceUnitDescriptors,
+            ImpliedBlockingPersistenceUnitTypeBuildItem impliedBlockingPersistenceUnitType,
+            List<JdbcDataSourceBuildItem> jdbcDataSources, // just make sure the datasources are initialized
+            Capabilities capabilities,
+            BuildProducer<AdditionalBeanBuildItem> additionalBeans,
+            BuildProducer<SyntheticBeanBuildItem> syntheticBeanBuildItemBuildProducer) {
+        if (persistenceUnitDescriptors.isEmpty()) {
+            // No persistence units have been configured so bail out
+            return;
+        }
+
+        for (PersistenceUnitDescriptorBuildItem persistenceUnitDescriptor : persistenceUnitDescriptors) {
+            String persistenceUnitName = persistenceUnitDescriptor.getPersistenceUnitName();
+            String persistenceUnitConfigName = persistenceUnitDescriptor.getConfigurationName();
```

Comment:
With your changes, the concept of "configuration name" might not be relevant at all anymore... If so we'll need to remember to clean this up and remove `getConfigurationName()` and related code from the codebase.